test/e2e/task_start_e2e_test.go

```diff
@@ -0,0 +1,147 @@
+		res := icmd.RunCmd(icmd.Cmd{Command: []string{"kubectl", "apply", "-f", TestResourcePath("buildah.yaml"), "-n", namespace}, Timeout: timeout})
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+		})
+		time.Sleep(1 * time.Second)
+	})
+
+	t.Run("Get list of Tasks from namespace  "+namespace, func(t *testing.T) {
+		res := icmd.RunCmd(run("task", "list", "-n", namespace))
+
+		expected := ListAllTasksOutput(t, c, map[int]interface{}{
+			0: &TaskData{
+				Name: "buildah",
+			},
+		})
+
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+			Out:      expected,
+		})
+	})
+
+	t.Run("create buildah resources in namespace "+namespace, func(t *testing.T) {
+		t.Logf("Creating Git PipelineResource")
+		if _, err := c.PipelineResourceClient.Create(getGitResource(namespace)); err != nil {
+			t.Fatalf("Failed to create Pipeline Resource `%s`: %s", "nocode", err)
+		}
+
+		t.Logf("Creating Image PipelineResource")
+		if _, err := c.PipelineResourceClient.Create(getImageResource(repo, namespace)); err != nil {
+			t.Fatalf("Failed to create Image Resource `%s`: %s", "image", err)
+		}
+
+	})
+
+	t.Run("Start Task Run using task start command with SA as 'pipeline' ", func(t *testing.T) {
+
+		res := icmd.RunCmd(run("task", "start", "buildah",
+			"-i=source=nocode",
+			"-o=image=image",
+			"-p=TLSVERIFY=false",
+			"-n", namespace))
+
+		time.Sleep(3 * time.Second)
+
+		vars := make(map[string]interface{})
+
+		taskRunGeneratedName := GetTaskRunListWithName(c, "buildah").Items[0].Name
+		vars["Namespace"] = namespace
+		vars["Taskrun"] = taskRunGeneratedName
+		expected := ProcessString(`(Taskrun started: {{.Taskrun}}
+
+In order to track the taskrun progress run:
+tkn taskrun logs {{.Taskrun}} -f -n {{.Namespace}}.*)`, vars)
+
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+		})
+		assert.Assert(t, is.Regexp(expected, res.Stdout()))
+
+		t.Logf("Waiting for TaskRun in namespace %s to Pass", namespace)
+		if err := WaitForTaskRunState(c, taskRunGeneratedName, TaskRunSucceed(taskRunGeneratedName), "TaskRunSucceed"); err != nil {
+			t.Errorf("Error waiting for TaskRun to finish: %s", err)
+		}
+
+	})
+
+	t.Run("Get list of Taskruns from namespace  "+namespace, func(t *testing.T) {
+		time.Sleep(1 * time.Second)
+
+		res := icmd.RunCmd(run("taskrun", "list", "-n", namespace))
+
+		expected := ListAllTaskRunsOutput(t, c, false, map[int]interface{}{
+			0: &TaskRunData{
+				Name:   "buildah-run-",
+				Status: "Succeeded",
+			},
+		})
+
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+			Out:      expected,
+		})
+
+	})
+
+}
+
+func getGitResource(namespace string) *v1alpha1.PipelineResource {
+	return tb.PipelineResource("nocode", namespace, tb.PipelineResourceSpec(
+		v1alpha1.PipelineResourceTypeGit,
+		tb.PipelineResourceSpecParam("url", "https://github.com/kelseyhightower/nocode"),
```

Comment:
We should not use this repo. Could we use a more legitimate sample application here?

Reply:
We are using the same in catalog also https://github.com/tektoncd/catalog/blob/v1beta1/buildah/tests/run.yaml#L18 so may be fine

Reply:
And we don't need complex tasks like buildah for our e2e, a simple task which has a git resource and param as filename- > and do cat filename is enough for our e2e I think, as we are testing the creation, not execution.

Reply:
@piyush-garg updated task as per your comments.

Reply:
I would prefer not using no code still. 

I would also actually prefer the original buildah Task. We should have something more complex as we are not gaining much insight out of using a hello world kind of Task. This can also serve as a way to dog food pipelines in ways. I think we gain a lot of value for not just the CLI with testing certain aspects of execution.